### PR TITLE
reorganize configuration syntax intro

### PR DESCRIPTION
**Pull request by moxious**
This PR attempts to do a couple of things:

* Contextualize the configuration guidance by making the point upfront that what configuration syntax does is specify a pipeline of connected components
* Give a simple fully-worked example up top of all concepts and how they work together, with comments
* Re-organize the page to chain the concepts (block, expression, attribute) in the order they're introduced in

@rfratto this is another trial balloon along the lines we discussed when we met. More than open to feedback here, but after that discussion hopefully it'll make sense where I'm coming from.  The simplest possible worked example that would show off the "full conceptual scope" of config (by this I mean: collection, transformation, delivery) is a good thing to have up top.

Nobody's going to use the referenced example, so its particulars don't matter much beyond the KISS principle. (This example happens to be lifted from a video we did on reporting logs to loki).  It's there to show the concepts in action before we get into a more technical discussion down the page about syntactically "what is a block".

There was an earlier example block that I removed because it was showing off "syntax only" before the concept of a block had been introduced.

### CODE REVIEW

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (rfratto) [inline docs/sources/concepts/configuration-syntax/_index.md:66]: Are we concerned about any potential confusion now that blocks (which are composed of attributes) are explained before attributes?

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (rfratto) [inline docs/sources/concepts/configuration-syntax/_index.md:131]: Likely not in scope for this PR, but I wonder if this content adds anything new and if it makes more sense to merge it into other headers. 

WDYT about the long-term value of this section @moxious?

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (rfratto) [inline docs/sources/concepts/configuration-syntax/_index.md:21]: `local.file_match` is a weird one, as it fits into the same bucket as the `discovery` components. It's not collecting any data, but performs discovery to enrich other components.

Do you think this distinction is useful at the introduction level? If so, maybe it could be a fourth category of components so the categories become:

1. Data enrichment (local.file_match, discovery.kubernetes, etc.) 
2. Collection 
3. Transformation
4. Delivery 

I'm ok with either; I'll defer to your judgment for what makes the most sense for high-level introductory content.

**Review by rfratto — APPROVED** [review]
High level LGTM, pending a review from @clayton-cornell for grammar/style/etc. I left a few questions but I'm fine with this being merged. Thanks!

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (rfratto) [reply docs/sources/concepts/configuration-syntax/_index.md:66]: If this is a concern, but the ordering is still preferred, maybe we can add some kind of teaser to explain attributes, something like: 

> You use _Blocks_ to configure components and groups of settings called attributes.

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (clayton-cornell) [inline docs/sources/concepts/configuration-syntax/_index.md:65]:
```suggestion

```

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (clayton-cornell) [inline docs/sources/concepts/configuration-syntax/_index.md:59]:
```suggestion
The {{< param "PRODUCT_NAME" >}} syntax aims to reduce errors in configuration files by making configurations easier to read and write.
The {{< param "PRODUCT_NAME" >}} syntax uses blocks, attributes, and expressions.
The blocks can be copied and pasted from the documentation to help you get started as quickly as possible.
```

The sentence about blocks, attributes, and expressions was out of place in the flow. I moved it up here, and suggest deleting it from line 64. We might need to look into how lines 61 and 62 fit into the flow of ideas here too...

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (clayton-cornell) [reply docs/sources/concepts/configuration-syntax/_index.md:66]: I think this suggestion is more clear and a good replacement for line 68.

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (clayton-cornell) [inline docs/sources/concepts/configuration-syntax/_index.md:119]:
```suggestion
{{< param "PRODUCT_NAME" >}} is:
```

We could simplify this down to a more direct statement

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (clayton-cornell) [inline docs/sources/concepts/configuration-syntax/_index.md:123]:
```suggestion
* _Fast_: The configuration language is fast, so the component controller can quickly evaluate changes.
* _Simple_: The configuration language is easy to read and write to minimize the learning curve.
* _Easy to debug_: The configuration language gives detailed information when there's a mistake in the configuration file.
```

Simplify to be more direct

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (clayton-cornell) [reply docs/sources/concepts/configuration-syntax/_index.md:131]: I was thinking the same. These points feel like a repetition or a rephrase of previous info... and it might be better to find a way to rephrase and fit it into the content around lines 59-65... somewhere in that area.  Might help to commit some of the other changes and do a `make docs` so this can all be read in the rendered output.

Comment (moxious) [reply docs/sources/concepts/configuration-syntax/_index.md:131]: Noting the goals of the design for a configuration syntax is, in my view - of primary interest to the people who want to hack on the syntax or contribute to core alloy.  So it's pretty advanced material.  That's why I chose to locate it at the bottom.  It wasn't in the scope of this PR to try to remove something, but being frank - if you're a beginning user you probably won't read this, because just knowing how to use the config syntax is enough.

If I were to move this content, I would move it into a section that's about "how to contribute to alloy" or "why we did what we did".  Very useful information to gate what kind of contributions would be welcome.

File: docs/sources/concepts/configuration-syntax/_index.md
Comment (moxious) [reply docs/sources/concepts/configuration-syntax/_index.md:21]: I borrowed this from a worked example my team did, I don't have anything wrapped up in its specifics.

If we can keep the example technically correct while making it shorter & simpler, that's a win.  We specifically don't need to teach anything advanced here, it's only there to demonstrate the "pipeline/flow" concept with a simple example anyone can follow (report logs)

**Review by clayton-cornell — APPROVED** [review]
This looks like a nice improvement :-)

**Pull request merged**
